src/features/serverTasks/serverTaskWaiter.ts

```diff
@@ -9,62 +9,76 @@ export class ServerTaskWaiter {
         serverTaskIds: string[],
         statusCheckSleepCycle: number,
         timeout: number,
-        pollingCallback?: (serverTaskDetails: ServerTaskDetails) => void
-    ): Promise<PromiseSettledResult<ServerTask | null>[]> {
+        pollingCallback?: (serverTask: ServerTask) => void
+    ): Promise<ServerTask[]> {
         const spaceServerTaskRepository = new SpaceServerTaskRepository(this.client, this.spaceName);
-        const taskPromises: Promise<ServerTask | null>[] = [];
-        for (const serverTaskId of serverTaskIds) {
-            taskPromises.push(this.waitForTask(spaceServerTaskRepository, serverTaskId, statusCheckSleepCycle, timeout, pollingCallback));
-        }
-        return await Promise.allSettled(taskPromises);
+
+        return this.waitForTasks(spaceServerTaskRepository, serverTaskIds, statusCheckSleepCycle, timeout, pollingCallback);
     }
 
     async waitForServerTaskToComplete(
         serverTaskId: string,
         statusCheckSleepCycle: number,
         timeout: number,
-        pollingCallback?: (serverTaskDetails: ServerTaskDetails) => void
-    ): Promise<ServerTask | null> {
+        pollingCallback?: (serverTask: ServerTask) => void
+    ): Promise<ServerTask> {
         const spaceServerTaskRepository = new SpaceServerTaskRepository(this.client, this.spaceName);
-        return this.waitForTask(spaceServerTaskRepository, serverTaskId, statusCheckSleepCycle, timeout, pollingCallback);
+        const tasks = await this.waitForTasks(spaceServerTaskRepository, [serverTaskId], statusCheckSleepCycle, timeout, pollingCallback);
+        return tasks[0];
     }
 
-    private async waitForTask(
+    private async waitForTasks(
         spaceServerTaskRepository: SpaceServerTaskRepository,
-        serverTaskId: string,
+        serverTaskIds: string[],
         statusCheckSleepCycle: number,
         timeout: number,
-        pollingCallback?: (serverTaskDetails: ServerTaskDetails) => void
-    ): Promise<ServerTask | null> {
+        pollingCallback?: (serverTask: ServerTask) => void
+    ): Promise<ServerTask[]> {
         const sleep = async (ms: number) => new Promise((r) => setTimeout(r, ms));
 
         let stop = false;
         const t = setTimeout(() => {
             stop = true;
         }, timeout);
 
+        const completedTasks: ServerTask[] = [];
+
         while (!stop) {
             try {
-                if (pollingCallback) {
-                    const taskDetails = await spaceServerTaskRepository.getDetails(serverTaskId);
-                    pollingCallback(taskDetails);
+                const tasks = await spaceServerTaskRepository.getByIds(serverTaskIds);
```

Comment:
We should only kick this off if serverTaskIds length is > 0. Should not matter too much from the way getByIds is implemented but as a safety check as no task ids means all for server

Reply:
Good pickup, I'd forgotten about that bit. I've added an early exit at the top of the method to prevent the first call possibly having 0 ids. The if at the bottom of the loop will set `stop` to true and therefore already prevent it from happening on subsequent calls.